docs/src/lib/types.md

```diff
@@ -57,6 +57,52 @@ The `RepeatedVector` and `StackedVector` types are subtypes of `AbstractVector`
 with the exception that they are read only. Note that they are not exported and should not be constructed directly,
 but they are columns of a `DataFrame` returned by `stackdf` and `meltdf`.
 
+## [The design of handling of columns of a `DataFrame`](@id man-columnhandling)
+
+When a `DataFrame` is constructed columns are copied by default. You can disable
+this behavior by setting `copycols` keyword argument to `false`. The exception is
+if an `AbstractRange` is passed as a column, then it is always collected to a `Vector`.
+
+Also functions that transform a `DataFrame` to produce a new `DataFrame` perform a copy of the columns,
+unless they are passed `copycols=false` (available only for functions
+that could perform a transformation without copying the columns). Examples of such functions are [`vcat`](@ref),
+[`hcat`](@ref), [`filter`](@ref), [`dropmissing`](@ref), [`join`](@ref), `getindex`,
+[`copy`](@ref) or the [`DataFrame`](@ref) constructor mentioned above.
+
+On the contrary, functions that create a view of a `DataFrame` *do not* by definition make copies of
+the columns, and therefore require particular caution. This includes `view`, which returns
+a `SubDataFrame` or a `DataFrameRow`, and `groupby`, which returns a `GroupedDataFrame`.
+
+A partial exception to this rule are the [`stackdf`](@ref) and [`meltdf`](@ref) functions which
+create a `DataFrame` that contains views of the columns from the source `DataFrame`.
+
+In-place functions whose names end with `!` (like `sort!` or [`dropmissing!`](@ref),
+`setindex!`, `push!`, `append!`) may mutate the column vectors of the `DataFrame` they take
+as an argument. These functions are safe to call due to the rules described above,
+*except* when a view of the `DataFrame` is in use (via a `SubDataFrame`, a `DataFrameRow`
+or a `GroupedDataFrame`). In the latter case, the view might become corrupted,
+which make trigger errors, silently return invalid data or even cause Julia to crash.
```

Comment:
my first (incorrect) reading of the paragraph above was that it was not safe to perform those on views.
But those are in-fact most (entirely?) operations that are not possible to perform on views.

Is there an extra sentence that could be added to make it clear that this is mostly (entirely?)
talking about that mutating the original(i.e. parent) dataframe while a view of it exists might corrupt the view

Reply:
I have added some more explanation